include/libdnf/advisory/advisory_reference.hpp

```diff
@@ -40,8 +38,8 @@ class AdvisoryReference {
 
     /// Get type of this reference.
     ///
-    /// @return Type of this reference.
-    Type get_type() const;
+    /// @return type of this reference as std::string.
+    std::string get_type() const;
```

Comment:
> Looks fine to me in this case to use strings instead of an enum. Though I'm not sure why `AdvisoryQuery` needs to accept the types as strings as you state in the commit description.

I remember we discussed it with @j-mracek and I had a TODO for it but to be honest I kind of forgot the reasoning. :disappointed: 

> Also with this change I think we need to document the list of possible type values in the documentation. I think either in multiple places or (better) in a single place to which we link from those multiple places (i.e. for advisory type at least here in the `get_type()`'s docstring and then in `AdvisoryQuery::filter_type()`).

I have added a commit for it, adding it in multiple places for now because I am not sure how to make a link here. Do you mean just writing something like: "Possible types are described in the `get_type()` method."?

Reply:
I'm also not sure how to add a link, there's probably a way in Doxygen... I'm not sure myself if to link to a single place or list the types in multiple places. Listing in multiple places might actually be more straightforward and easier to read in the docs, so I think it's just fine the way you did it.